runtime/commonMain/src/kotlinx/serialization/internal/Enums.kt

```diff
@@ -66,6 +54,14 @@ open class CommonEnumSerializer<T>(val serialName: String, val choices: Array<T>
             { "$index is not among valid $serialName choices, choices size is ${choices.size}" }
         return choices[index]
     }
+
+    companion object {
+        /**
+         * Factory method for easier instantiation from user/test code.
+         */
+        inline fun <reified E: Enum<E>> create(serialName: String): CommonEnumSerializer<E> =
+            CommonEnumSerializer(serialName, enumValues<E>(), enumValues<E>().map { it.name }.toTypedArray())
```

Comment:
1) Unnecessary type argument in `enumValue`
2) This API is only used in tests, thus probably shouldn't be public
3) More Kotlin-ish API would be constructor-like functions:  `inline fun <reified E: Enum<E>>CommonEnumSerializer(serialName: String)` without companion

Reply:
It would be convenient if someone would be willing to create this serializer manually. However, I'm not sure that this use-case will be very popular

Reply:
Then it's better not to introduce it in public API I suppose